yarn-project/simulator/src/avm/fixtures/avm_simulation_tester.ts

```diff
@@ -0,0 +1,96 @@
+import { GasFees, GlobalVariables } from '@aztec/circuits.js';
+import { type ContractArtifact } from '@aztec/foundation/abi';
+import { encodeArguments } from '@aztec/foundation/abi';
+import { AztecAddress } from '@aztec/foundation/aztec-address';
+import { Fr } from '@aztec/foundation/fields';
+import { openTmpStore } from '@aztec/kv-store/lmdb';
+import { AvmTestContractArtifact } from '@aztec/noir-contracts.js/AvmTest';
+import { MerkleTrees } from '@aztec/world-state';
+
+import { type AvmContractCallResult } from '../../avm/avm_contract_call_result.js';
+import {
+  getContractFunctionArtifact,
+  getFunctionSelector,
+  initContext,
+  initExecutionEnvironment,
+  resolveContractAssertionMessage,
+} from '../../avm/fixtures/index.js';
+import { PublicEnqueuedCallSideEffectTrace } from '../../public/enqueued_call_side_effect_trace.js';
+import { WorldStateDB } from '../../public/public_db_sources.js';
+import { AvmPersistableStateManager, AvmSimulator } from '../../server.js';
+import { BaseAvmSimulationTester } from './base_avm_simulation_tester.js';
+import { SimpleContractDataSource } from './simple_contract_data_source.js';
+
+const TIMESTAMP = new Fr(99833);
+const DEFAULT_GAS_FEES = new GasFees(2, 3);
+
+export class AvmSimulationTester extends BaseAvmSimulationTester {
+  constructor(
+    contractDataSource: SimpleContractDataSource,
+    merkleTrees: MerkleTreeWriteOperations,
+    skipContractDeployments = false,
+    private stateManager: AvmPersistableStateManager,
+  ) {
+    super(contractDataSource, merkleTrees, skipContractDeployments);
+  }
+
+  static async create(skipContractDeployments = false): Promise<AvmSimulationTester> {
+    const contractDataSource = new SimpleContractDataSource(skipContractDeployments);
+    const merkleTrees = await (await MerkleTrees.new(openTmpStore())).fork();
+    const worldStateDB = new WorldStateDB(merkleTrees, contractDataSource);
+    const trace = new PublicEnqueuedCallSideEffectTrace();
+    const firstNullifier = new Fr(420000);
+    // FIXME: merkle ops should work, but I'm seeing frequent (but inconsistent) bytecode retrieval
+    // failures on 2nd call to simulateEnqueuedCall with merkle ops on
+    const stateManager = await AvmPersistableStateManager.create(
+      worldStateDB,
+      trace,
+      /*doMerkleOperations=*/ false,
+      firstNullifier,
+    );
+    return new AvmSimulationTester(contractDataSource, merkleTrees, skipContractDeployments, stateManager);
+  }
+
+  async simulateEnqueuedCall(
+    fnName: string,
+    args: any[],
+    address: AztecAddress = this.getFirstContractInstance().address,
+    sender = AztecAddress.random(),
+    contractArtifact: ContractArtifact = AvmTestContractArtifact,
+    isStaticCall = false,
+  ): Promise<AvmContractCallResult> {
+    const fnSelector = getFunctionSelector(fnName, contractArtifact);
+    const fnAbi = getContractFunctionArtifact(fnName, contractArtifact);
+    const encodedArgs = encodeArguments(fnAbi!, args);
```

Comment:
ah, now I understand, it's the actual args to the "noir function" not the calldata in field format. Docs needed :)